src/client/ws.rs

```diff
@@ -0,0 +1,56 @@
+use std::io::{self, Write};
+use std::time::Duration;
+use std::{thread, process};
+use futures::stream::SplitStream;
+use futures;
+use futures_util::StreamExt;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::sync::mpsc;
+use tokio_tungstenite::WebSocketStream;
+use tokio_tungstenite::tungstenite::Error;
+use tokio_tungstenite::{connect_async, tungstenite::protocol::Message};
+
+pub async fn connect(url: url::Url) -> WebSocketStream<tokio_tungstenite::MaybeTlsStream<tokio::net::TcpStream>> {
+    let (ws, _) = connect_async(url).await.expect("Failed to connect");
+    println!("WebSocket connection to server established");
+    ws
+}
+
+pub async fn response(mut ws_rcv: SplitStream<WebSocketStream<tokio_tungstenite::MaybeTlsStream<tokio::net::TcpStream>>>)
+{
+    // websocket reading
+    while let Some(result) = ws_rcv.next().await {
+        let msg = match result {
+            Ok(msg) => msg,
+            Err(e) => {
+                eprintln!("error receiving message: {}", e);
+                break;
+            }
+        };
+        let mut data = msg.into_data();
+        data.extend_from_slice("\n".as_bytes().as_ref());
+        tokio::io::stdout().write(&data).await.unwrap();
```

Comment:
```suggestion
        tokio::io::stdout().write(&data).await.unwrap();
```
`?` statt `.unwrap()`?